### DISCUSSION TITLE
enqueueOrIgnore ?

### DISCUSSION THREAD

**Discussion by jhanzel-be** [jobrunr/jobrunr#752] [Q&A] [answered]
Hi,

We have recently implemented the eqneueuOrReplace to avoid duplicating a set of jobs that runs on an hourly basis.  Sometimes the jobs do not complete before the next hour and we want to avoid running them unnecessarily.  In our case, the jobs in question are scheduled 30 seconds apart into the future. The problem we're running into is now the jobs are not getting duplicated (yay), but the newer job sometimes replaces the older job that is still scheduled and schedules it further into the future.  And in some cases if their is a larger number of jobs,  a few jobs get scheduled further and further out and do not get run in a timely way.

Are there any alternative or variations to the enqueueOrReplace functionality that would allow it to simply ignore or no-op scheduling if the job already exists ?  I do not want to replace the job.  Apologize in advance for not reviewing the API docs in more details.
Thanks,
Joey

**Comment by rdehuyss** [comment]
If you can pass in your own Job ID (so your own UUID), then they are ignored. You can find the documentation around that [here](https://www.jobrunr.io/en/documentation/faq/#im-listening-for-jobs-using-service-bus-messages-in-a-load-balanced-environment-and-i-want-to-schedule-jobs-only-once).

Would that help?

**Reply by jhanzel-be** [reply]
Hi Ron,
We're currently using our own JobID with scheduleOrReplace.   It's working too well.  What would happen if we passed in our own Job ID into schedule and a previous job was complete (succeeded or failed).  Would it still no-op ?    I'm looking for the no-op/skip when the job is incomplete (enqueued, scheduled or processing).   Would it work differently If we deleted our jobs after they completed and later scheduled with a previously deleted UUID ?

**Reply by rdehuyss** [reply]
Yes, if you call `StorageProvider.deletePermanently()` after the job completed successfully, you can then recreate it later with the same id. 

I think your issue relates to an outstanding feature request: https://github.com/jobrunr/jobrunr/issues/236

**Comment by jhanzel-be** [comment, marked as answer]
Thanks @rdehuyss.

I think the best outcome in my case is to no-op subsequent scheduleOrReplace call if the previously scheduled job has yet run.  I was thinking of doing some testing with the JobResultWithBackOffInfo isReady method and skipping the call to schedule if  the result is not Ready,  would that be an appropriate use of the isReady method?

We are providing our own JobId's so something like this...

  ...
    final UUID jobId = getUuid(key);
    final JobResultWithBackOffInfo job = getJobScheduler().getJobResult(jobId);
    if (job != null && !job.isReady()) {
      return null;
    }
    return getJobScheduler().scheduleOrReplace(jobId, instant, iocJob);
 ....

**Reply by rdehuyss** [reply]
Yes, that can work!
